src/compiler/transformers/esnext.ts

```diff
@@ -677,8 +677,8 @@ namespace ts {
             const trailingStatements = endLexicalEnvironment();
             if (statementOffset > 0 || some(statements) || some(trailingStatements)) {
                 const block = convertToFunctionBody(body, /*multiLine*/ true);
-                addRange(statements, block.statements.slice(statementOffset));
                 addRange(statements, trailingStatements);

```

Comment:
Given the change in position, `leadingStatements` would be a more appropriate term now.